### DISCUSSION TITLE
Support sharp's `withoutEnlargement` option

### DISCUSSION THREAD

**Discussion by MarcusCemes** [JonasKruckenberg/imagetools#257] [Ideas]
Creating presets with a fixed set of sizes at the moment will generate unnecessary sizes, is there any way to implement the `withoutEnlargement` option when resizing to avoid upscaling images, and possibly even remove duplicate images if they are constrained to the original image's dimensions?

Or is it possible to get access to the source image dimensions from within a custom `resolveConfigs` function in order to selectively add widths that would not cause upscaling or duplicate images?

This was one of my primary goals when I developed https://github.com/MarcusCemes/image-processing-pipeline, at the moment I'm lacking in time, I was hoping to pick something else off the shelf!

**Comment by JonasKruckenberg** [comment]
> Creating presets with a fixed set of sizes at the moment will generate unnecessary sizes, is there any way to implement the `withoutEnlargement` option when resizing to avoid upscaling images, and possibly even remove duplicate images if they are constrained to the original image's dimensions?

I like this idea, though I guess since imagetools has no support for relative resizing this isn't such a big deal?
It's quite easy to look up the original sizes.
But I agree this is a good safeguard to have.

> Or is it possible to get access to the source image dimensions from within a custom `resolveConfigs` function in order to selectively add widths that would not cause upscaling or duplicate images?

This is certainly possible yes.

> This was one of my primary goals when I developed https://github.com/MarcusCemes/image-processing-pipeline, at the moment I'm lacking in time, I was hoping to pick something else off the shelf!

Well what can I say, I'm incredibly humbled but also very much stripped for time too (you can probably tell)
So if you end up implementing this as a custom directive I'm happy to accept a PR to make it builtin 👍🏻
